### ISSUE TITLE
Trying to add OneDrive fails

### ISSUE THREAD

**Issue by vytux-com** [Kyrodan/KeeAnywhere#193]
When adding a Personal OneDrive account I get this error:

```
Authentication failed!

Entry point was not found.
Microsoft.Graph.Core
   at Microsoft.Graph.IHttpProvider.SendAsync(HttpRequestMessage request)
   at Microsoft.OneDrive.Sdk.Authentication.OAuthHelper.<SendTokenRequestAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.OneDrive.Sdk.Authentication.OAuthHelper.<RedeemAuthorizationCodeAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at KeeAnywhere.StorageProviders.OneDrive.OneDriveStorageConfigurator.<Claim>d__b.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at KeeAnywhere.OAuth2.OAuth2Form.<OnNavigated>d__3.MoveNext()
System.Threading.Tasks.Task`1[System.Net.Http.HttpResponseMessage] SendAsync(System.Net.Http.HttpRequestMessage)
```

**Comment by vytux-com** [comment]
![image](https://user-images.githubusercontent.com/5370583/70282896-72f00800-1813-11ea-9260-20f5dc50148f.png)
As can be seen here it logs in OK and was granted permissions, however it failed straight away after that. Keepass2Android implementation on the mobile device is working OK

**Comment by Kyrodan** [comment]
Is this problem still present? I cannot reproduce this?
Do you use or should use a proxy?

**Comment by vytux-com** [comment]
Yes still getting same error, upgraded to 1.6.0 same issue, no proxy and tried on several different networks. Will try on a different PC sometime this week, but I don't believe that is an issue.

'''
Authentication failed!

Entry point was not found.
Microsoft.Graph.Core
   at Microsoft.Graph.IHttpProvider.SendAsync(HttpRequestMessage request)
   at Microsoft.OneDrive.Sdk.Authentication.OAuthHelper.<SendTokenRequestAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.OneDrive.Sdk.Authentication.OAuthHelper.<RedeemAuthorizationCodeAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at KeeAnywhere.StorageProviders.OneDrive.OneDriveStorageConfigurator.<Claim>d__b.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at KeeAnywhere.OAuth2.OAuth2Form.<OnNavigated>d__3.MoveNext()
System.Threading.Tasks.Task`1[System.Net.Http.HttpResponseMessage] SendAsync(System.Net.Http.HttpRequestMessage)

**Comment by vytux-com** [comment]
OK so i tried on a different PC and that one worked, they are both on the same network, the one that didn't work is domain connected and the one that did is not.

I will see if I can find any other possible problems

**Comment by Mordeth0** [comment]
I think I have this same issue.  I used to use OneDrive directly on my work computer, but that was recently blocked.  I have a SOCKS5 proxy setup thru an SSH tunnel to a home computer.  I can configure Firefox, Chrome, etc to utilize the proxy and it works just fine.  I configured KeePass to utilize the proxy thru Tools --> Options --> Advanced --> Proxy (it is an SSH tunnel so it is 127.0.0.1 on port 443).

When I try to add my OneDrive Account, I'm prompted to enter my email, then authenticate with Microsoft Authenticator.  It appears successful, but I immediately get a "KeePass - Fatal Error" See attached.
![image](https://user-images.githubusercontent.com/77125330/103951192-db3ea680-5103-11eb-8c78-84048f2f0279.png)

The text that is copied to the clipboard is below.  I'm guessing that KeeAnywhere is not properly utilizing the proxy settings in KeePass and my work is still blocking my connection to OneDrive.  Thoughts?


Error Copied to Clipboard:

Authentication failed!

Code: generalException
Message: An error occurred sending the request.

Microsoft.Graph.Core
   at Microsoft.Graph.HttpProvider.<SendRequestAsync>d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Graph.HttpProvider.<SendAsync>d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.OneDrive.Sdk.Authentication.OAuthHelper.<SendTokenRequestAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.OneDrive.Sdk.Authentication.OAuthHelper.<RedeemAuthorizationCodeAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at KeeAnywhere.StorageProviders.OneDrive.OneDriveStorageConfigurator.<Claim>d__b.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at KeeAnywhere.OAuth2.OAuth2Form.<OnNavigated>d__3.MoveNext()
Void MoveNext()


Inner:
An error occurred while sending the request.
mscorlib
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Graph.HttpProvider.<SendRequestAsync>d__21.MoveNext()
Void Throw()
